src/OSPSuite.Core/Serialization/SimModel/Services/SimModelExporter.cs

```diff
@@ -52,6 +52,9 @@ private void exportToCode(IModelCoreSimulation modelCoreSimulation, string outpu
          var simModelXml = ExportSimModelXml(modelCoreSimulation, SimModelExportMode.Full);
          var simModelSimulation = new Simulation();
 
+         if (codeExportLanguage == CodeExportLanguage.Cpp)
+            simModelSimulation.Options.KeepXMLNodeAsString = true;
```

Comment:
For some weird reason cpp export relies on the original XML used for simulation creation. Per default XML string used to setup a SimModel-simulation is thrown away after simulation.Load. With the line above it's kept and can be accessed by the cpp export

Reply:
I do not understand what this line does, even with your comments

Reply:
I'd suggest to add the explanation in the code because it's just too weird

Reply:
done